src/AWSCredentials.jl

```diff
@@ -208,18 +208,27 @@ metadata is available (typically due to not running within an EC2 instance) then
 will be returned. See the AWS documentation for details on what metadata is available:
 https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
 
+If using KIAM or something similar which filters the calls to instance metadata,
+set the `ignored_codes` parameter to prevent exceptions being raised for certain statuses.
+
 # Arguments
 - `path`: The URI path to used to specify that metadata to return
+- `ignored_codes`: a tuple of HTTP status codes to ignore (default empty tuple)
 """
-function ec2_instance_metadata(path::AbstractString)
+
+function ec2_instance_metadata(path::AbstractString, ignored_codes::Tuple{Vararg{<:Integer}}=())
     uri = HTTP.URI(; scheme="http", host="169.254.169.254", path=path)
     request = try
-        @mock HTTP.request("GET", uri; connect_timeout=1)
+        Mocking.@mock HTTP.request("GET", uri; connect_timeout=1)
```

Comment:
The `@mock` macro is exported by `Mocking.jl`, no need to preface here.

```suggestion
        @mock HTTP.request("GET", uri; connect_timeout=1)
```

Reply:
Sorry, habit of mine